### PR TITLE
Add CreateOnlineStreamWithHotwords to C API

### PR DESCRIPTION
**Pull request by nmfisher**
Allows creating a new OnlineStream with hotwords via the C API.

Also added the __attribute__((visibility("default"))) modifier which makes it easier to use the C API as-is in FFI setups.

### CODE REVIEW

File: sherpa-onnx/c-api/c-api.cc [hunk 1]
```diff
@@ -107,6 +107,14 @@ SherpaOnnxOnlineStream *CreateOnlineStream(
   return stream;
 }
 
+SherpaOnnxOnlineStream *CreateOnlineStreamWithHotwords(
+    const SherpaOnnxOnlineRecognizer *recognizer, const char* hotwords) {
+  std::string hw(hotwords);
+  SherpaOnnxOnlineStream *stream =
+      new SherpaOnnxOnlineStream(recognizer->impl->CreateStream(hw));
```
Comment (csukuangfj) [inline sherpa-onnx/c-api/c-api.cc]:
```suggestion
SherpaOnnxOnlineStream *CreateOnlineStreamWithHotwords(
    const SherpaOnnxOnlineRecognizer *recognizer, const char *hotwords) {
  SherpaOnnxOnlineStream *stream =
      new SherpaOnnxOnlineStream(recognizer->impl->CreateStream(hotwords));
```

File: sherpa-onnx/c-api/c-api.h [hunk 2]
```diff
@@ -178,6 +178,13 @@ SHERPA_ONNX_API void DestroyOnlineRecognizer(
 SHERPA_ONNX_API SherpaOnnxOnlineStream *CreateOnlineStream(
     const SherpaOnnxOnlineRecognizer *recognizer);
 
+/// Create an online stream for accepting wave samples with the specified hot words.
+///
+/// @param recognizer  A pointer returned by CreateOnlineRecognizer()
+/// @return Return a pointer to an OnlineStream. The user has to invoke
+///         DestroyOnlineStream() to free it to avoid memory leak.
+SHERPA_ONNX_API SherpaOnnxOnlineStream *CreateOnlineStreamWithHotwords(const SherpaOnnxOnlineRecognizer *recognizer, const char* hotwords);
```
Comment (csukuangfj) [inline sherpa-onnx/c-api/c-api.h]:
```suggestion
SHERPA_ONNX_API SherpaOnnxOnlineStream *CreateOnlineStreamWithHotwords(const SherpaOnnxOnlineRecognizer *recognizer, const char *hotwords);
```

**Review by csukuangfj — COMMENTED** [review]
Thanks! Left a minor comment about style issues.

**Comment by nmfisher** [comment]
Done!

**Comment by csukuangfj** [comment]
Thank you for your contribution!

**Pull request merged**